Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/SensorBase.cs

```diff
@@ -51,9 +51,9 @@ public virtual byte[] GetCompressedObservation()
         }
 
         /// <inheritdoc/>
-        public virtual SensorCompressionType GetCompressionType()
+        public virtual CompressionSpec GetCompressionSpec()
         {
-            return SensorCompressionType.None;
+            return CompressionSpec.Default();
         }
```

Comment:
We cached the shapes previously, because they required allocating memory. I kept that pattern for the `ObservationSpec`s - it's not really necessary for performance (struct with InplaceArray, so no allocations), but it's not a bad idea since they shouldn't change at runtime. I don't think it's necessary for CompressionSpecs.